app/models/event_stream.rb

```diff
@@ -0,0 +1,38 @@
+class EventStream
+
+  PAGESIZE = 30
+  ROLLBACK = 2
+
+  def initialize(discussion_reader:, focused: nil)
+    @discussion_reader = discussion_reader
+    @discussion = discussion_reader.discussion
+    @focused = focused
+  end
+
+  def initial_events
+    Event.where(discussion: @discussion)
+         .where('sequence_id > ?', load_from_event)
+         .order(sequence_id: :asc)
+         .limit(PAGESIZE)
+  end
+
+  def focused_sequence_id
```

Comment:
I may just take this out, and have the client worry about focusing the right element.
The server will still need to know what sequence_id is being targeted in the anchor, though, so it can load the right events.